app/experimenter/docs/adrs/0003-analysis-visualization-architecture-approach.md

```diff
@@ -0,0 +1,49 @@
+# Analysis Visualization: Architecture Approach
+
+* Status: accepted
+* Author: Marina Samuel
+* Deciders: Nimbus Team
+* Date: 2020-09-16
+
+## Context and Problem Statement
+
+Analysis and visualization of experiment result data needs to be [part of the experiment console](https://github.com/mozilla/nimbus-shared/blob/main/docs/adr/0001-one-console-app.md). The mockups based on user interviews are [here](https://www.figma.com/file/q6Xh9g6LrOj2OQ59LbAFXm/Experiment-Dashboard-Mockup).
+
+Some decisions need to be made about how the data will be pulled out of [Jetstream](https://github.com/mozilla/jetstream) and visualized within the console. This ADR will highlight how this will be done.
+
+
+## Decision Drivers
+
+Drivers for fetching data:
+* Data is easy to fetch and minimizes the amount of new state added
+
+Drivers for visualizing data:
+* Reuse existing [Graph Paper](https://github.com/graph-paper-org/graph-paper) UI components if possible
+* How easy or possible is it to integrate Svelte components into React (since Graph Paper is in Svelte)
+
+
+## Considered Options
+
+**Fetching Data**
+1. Add a POST endpoint to Jetstream to send data over to Experiment Console. This data would be saved in the Django database and fetched from the front-end when needed using GraphQL.
+2. Add a REST API endpoint to Experiment Console which fetches the Jetstream data using GCS dependencies. These are fetched on-demand and not stored in the Django database.
+
+**Frontend**
+1. Use nvd3 or other graphing library as an alternative to Graph Paper
+2. Use Graph Paper/Svelte only for graphs, otherwise, all visualizations from the mockup will be in React
+3. Use Svelte for the entire visualization portion of the console
+
+## Decision Outcome
+
+**Fetching Data**
+Option (2) was chosen.
+
+While option (2) requires adding more state such as bucket credentials, it is more straight-forward than option (1). It does not require any changes on the Jetstream side to make POST calls. It also does not require storing the data to the Django database. This is ok since the aggregated data is very small, in the range of around 100kb per fetch.
+
+**Frontend**
+Option (2) was chosen.
+
+Option (1) is not ideal since existing libraries would be less customizable than our in-house library that we have already invested in.  The only reason to have chosen this was if it was not possible to embed Svelte in React. However, this has been proven to be possible.
```

Comment:
Did we also look into what specifically we can use here? If not it might be worth linking to a few potentials saying it's possible but the specific package is TBD.

Reply:
Do you mean to add links to potential graphing libraries we could use? Perhaps I should add that in the section above for "considered options"?

Reply:
Yeah, to help demonstrate it's possible and an integration option exists that works for our use case, but I'm not sure it's worth adding in "considered options" unless we've kind of fleshed out which one we want to go with. I was thinking something like:

>  The only reason to choose this option would be if it was not possible to embed Svelte in React, however, this is possible a package like [`react-svelte`](https://github.com/Rich-Harris/react-svelte).

... I was going to list a couple, but this is the only one I see after a quick Google search.

Reply:
It's actually already done in this PR: https://github.com/mozilla/experimenter/pull/3419. I'll add a sentence or two to describe the method used there.